### DISCUSSION TITLE
Why did this third party JS library not import correctly using AssetMapper?

### DISCUSSION THREAD

**Discussion by pimhakkert** [symfony/symfony#59326] [Q&A]
Hello, this is my first time using importmaps and I'm having a difficult time getting certain third party libraries to work. One I really struggled with was [tom-select](https://github.com/orchidjs/tom-select). I imported the package, but nothing happened. After trying for a while I attempted to directly link to the jsDeliver link (`https://cdn.jsdelivr.net/npm/tom-select/dist/js/tom-select.complete.min.js`) which seemed to do the job. But when I import `tom-select`, it no longer works. Is this a compiling issue by jsDeliver?

**Comment by MatTheCat** [comment]
Hello,

just tried to require tom-select and everything is working as expected so the issue must be on your side.

1. Run `bin/console importmap:require tom-select`. This will download the required assets and reference them in your `importmap.php`
2. Create or update your entry point so that it uses tom-select, e.g.
```js
// app.js

import TomSelect from "tom-select";

console.log(TomSelect);
```
3. Reference said entrypoint in your import map. It should return something like

```php
[
    'app' => [
        'path' => 'app.js',
        'entrypoint' => true,
    ],
    'tom-select' => [
        'version' => '2.4.1',
    ],
    '@orchidjs/sifter' => [
        'version' => '1.1.0',
    ],
    '@orchidjs/unicode-variants' => [
        'version' => '1.1.2',
    ],
    'tom-select/dist/css/tom-select.default.min.css' => [
        'version' => '2.4.1',
        'type' => 'css',
    ],
]
```
4. Add `{{ importmap('app') }}` in your template so that your entrypoint runs.

After loading the corresponding page, you should see something like `class q { constructor(t, i) }` in your browser console. That’s the `TomSelect` class, so everything is working :tada: 

> [!NOTE] 
> You could also be using [Symfony UX Autocomplete](https://ux.symfony.com/autocomplete).
